docs/add-runtime.md

```diff
@@ -0,0 +1,254 @@
+The build script is always executed during the build stage of tag deployment.
+
+The `launch.sh` file for an interpreted runtime should extract the `/tmp/code.tar.gz` file that contains both the user's code and the dependencies. This tarball was created by OpenRuntimes from the `/usr/code` folder and should install the dependencies that were pre-installed by the build stage and move them into the relevant locations for that runtime. It will then run the server ready for execution.
+
+---
+Compiled Languages only have a `build.sh` file.
+The `build.sh` script for a compiled runtime is used to move the user's source code and rename it into source files for the runtime (The `ENTRYPOINT_NAME` environment variable can help with this) it will also build the code and move it into the `/usr/code` folder. Compiled runtime executables **must** be called `runtime` for the ubuntu or alpine images to detect and run them.
+
+#### Note:
+`/tmp/code.tar.gz` is always created from the `/usr/code` folder in the build stage. If you need any files for either compiled or interpreted runtimes you should place them there and extract them from the `/tmp/code.tar.gz` during the `launch.sh` script to get the files you need.
+
+### 2.3 Writing the runtime
+Internally the runtime can be anything you like as long as it follows the standards set by the other runtimes.
+
+The best way to go about writing a runtime is like so:
+Initialize a web server that runs on port 3000 and uses any IP Address (0.0.0.0) and on each `POST` request do the following:
+1. Check that the `x-internal-challenge` header matches the `INTERNAL_RUNTIME_KEY` environment variable. If not return an error with a `401` status code and an `unauthorized` error message.
+2. Decode the executor's JSON POST request. This normally looks like so:
+```json
+{
+ "path": "/usr/code",
+ "file": "index.js",
+ "env": {
+         "hello":"world!"
+    },
+ "payload":"An Example Payload",
+ "timeout": 10
+}
+```
+For a compiled language you can disregard the `path` and `file` attribute if you like,
+
+`timeout` is also an optional parameter to deal with, if you can handle it please do. Otherwise, it doesn't matter since the connection will simply be dropped by the executor.
+
+You must create two classes for users to use within their scripts. A `Request` Class and a `Response` class
+The `Request` class must store `env`, `payload` and `headers` and pass them to the user's function. 
+The Request always goes before the response in the user's function parameters.
+The `Response` class must have two functions. 
+- A `send(string)` function which will return text to the request
+- and a `json(object)` function which will return JSON to the request setting the appropriate headers
```

Comment:
> You must create two classes for users to use within their scripts. A `Request` Class and a `Response` class

Not sure this is accurate . Not all of the runtimes use a Request and response class. Languages like JS just pass a native JS object with those properties.